content/1_General/Choosing_Lang.mdx

```diff
@@ -332,9 +332,54 @@ while lo != hi:
 print(-1 if lo == 0 else all_edges[lo-1][0])
 ```
 
-We are not sure whether it is possible to modify the above approach to receive
-full credit (please let us know if you manage to succeed)! Of course, it is
-possible to pass this problem in Python using DSU (a Gold topic):
+It is possible to optimise this approach even further to pass all testcases. This takes around 3.7s to run.
+
+```py
+def main():
+	f = open("wormsort.in","rb")
+	n,m = map(int, f.readline().split())
+	loc = [*map(int, f.readline().split())]
+	edges = [[] for _ in range(n)]
+	weights = []
+	lo,hi = 0, m - 1
```

Comment:
would suggest moving this line to just before `lo` and `hi` are used

Reply:
fixed